conantools.cmake

```diff
@@ -91,6 +99,10 @@ function(detect_host_profile output_file)
     else()
         set(_FN ${output_file})
     endif()
+
+    string(APPEND PROFILE "[conf]\n")
+    string(APPEND PROFILE "tools.cmake.cmaketoolchain:generator=${CMAKE_GENERATOR}\n")
```

Comment:
Is CMAKE_GENERATOR always defined as a variable? Even when not provided in CLI and defaulted?

Reply:
yup, should be defined by CMake itself, even when not provided by the user